devicetypes/fibargroup/fibaro-door-window-sensor-2.src/fibaro-door-window-sensor-2.groovy

```diff
@@ -129,12 +131,20 @@ def updated() {
 	}
 	logging("${device.displayName} - Executing updated()","debug")
 
-	if ( settings.temperatureHigh as Integer == 0 && settings.temperatureLow as Integer == 0 ) { 
-		sendEvent(name: "temperatureAlarm", value: null, displayed: false) 
-	} else if ( settings.temperatureHigh != null || settings.temperatureHigh != null ) {
-		sendEvent(name: "temperatureAlarm", value: "clear", displayed: false) 
+	def tempAlarmMap = ["clear": "cleared", "overheat": "heat", "underheat": "freeze"]
+	// Convert old device specific temperatureAlarm event to standard Temperature Alarm capability event
+	def temperatureAlarmCurrentValue = device.currentValue("temperatureAlarm")
+	if (tempAlarmMap.containsKey(temperatureAlarmCurrentValue)) {
+		sendEvent(name: "temperatureAlarm", value: tempAlarmMap[temperatureAlarmCurrentValue], displayed: true)
 	}
-	
+
+	def currentTemperature = device.currentValue("temperature")
+	def temperatureHigh = settings.temperatureHigh ? new BigDecimal(settings.temperatureHigh) : null
+	def temperatureLow = settings.temperatureLow ? new BigDecimal(settings.temperatureLow) : null
+	if ((temperatureHigh != null && (currentTemperature < temperatureHigh)) || (temperatureLow != null && (currentTemperature > temperatureLow) )) {
+		sendEvent(name: "temperatureAlarm", value: "cleared", displayed: false)

```

Comment:
ideally this would only be sent if the temperature alarm isn't already clear

Reply:
right, fix is ready